packages/wdio-browserstack-service/src/instrumentation/performance/performance-tester.ts

```diff
@@ -0,0 +1,329 @@
+import { performance, PerformanceObserver } from 'node:perf_hooks'
+import util from 'node:util'
+import worker from 'node:worker_threads'
+import path from 'node:path'
+import { arch, hostname, platform, type, version } from 'node:os'
+
+import got from 'got'
+
+import { BStackLogger } from '../../bstackLogger.js'
+import { EDS_URL, PERF_MEASUREMENT_ENV } from '../../constants.js'
+
+type PerformanceDetails = {
+    success?: true,
+    failure?: string,
+    testName?: string,
+    worker?: string | number,
+    command?: string,
+    hookType?: string,
+    platform?: string | number
+}
+
+export default class PerformanceTester {
+    static _observer: PerformanceObserver
+    static _csvWriter: any
+    private static _events: PerformanceEntry[] = []
+    private static _measuredEvents: PerformanceEntry[] = []
+    static started = false
+    static details: {[key: string]: PerformanceDetails} = {}
+    static eventsMap: {[key: string]: number} = {}
+    static browser?: WebdriverIO.Browser
+    static scenarioThatRan: string[]
+    static jsonReportDirName = 'performance-report'
+    static jsonReportDirPath = path.join(process.cwd(), 'logs', this.jsonReportDirName)
+    static jsonReportFileName = `${this.jsonReportDirPath}/performance-report-${PerformanceTester.getProcessId()}.json`
+
+    static startMonitoring(csvName: string = 'performance-report.csv') {
+        // Create performance-report dir if not exists already
+        if (!fs.existsSync(this.jsonReportDirPath)) {
+            fs.mkdirSync(this.jsonReportDirPath, { recursive: true })
+        }
+        this._observer = new PerformanceObserver(list => {
+            list.getEntries().forEach(entry => {
+                let finalEntry = entry
+                if (entry.entryType === 'measure') {
+                    finalEntry = entry.toJSON()
+                    if (this.details[entry.name]) {
+                        finalEntry = Object.assign(finalEntry, this.details[entry.name])
+                    }
+
+                    delete this.details[entry.name]
+                    this._measuredEvents.push(finalEntry)
+                } else if (process.env[PERF_MEASUREMENT_ENV]) {
+                    this._events.push(finalEntry)
+                }
+            })
+        })
+        const entryTypes: EntryType[] = ['measure']
+        if (process.env[PERF_MEASUREMENT_ENV]) {
+            entryTypes.push('function')
+        }
+        this._observer.observe({ buffered: true, entryTypes: entryTypes })
+        this.started = true
+        if (process.env[PERF_MEASUREMENT_ENV]) {
+            this._csvWriter = createObjectCsvWriter({
+                path: csvName,
+                header: [
+                    { id: 'name', title: 'Function Name' },
+                    { id: 'time', title: 'Execution Time (ms)' }
+                ]
+            })
+        }
+    }
+
+    static getPerformance() {
+        return performance
+    }
```

Comment:
Let's not export a Node.js primitive as static class function, just import it where you need it